pkg/vm/vm_test.go

```diff
@@ -2083,6 +2083,43 @@ func TestUNPACKBadNotArray(t *testing.T) {
 	runWithArgs(t, prog, nil, 1)
 }
 
+func TestPACKMAPDuplicateKeys(t *testing.T) {
+	prog := makeProgram(opcode.PACKMAP)
+	vm := load(prog)
+
+	keys := []string{"duplicateKey", "uniqueKey", "duplicateKey", "anotherUniqueKey"}
+	values := []string{"value1", "value2", "value3", "value4"}
+
+	for i := range keys {
+		vm.estack.PushVal(values[i])
+		vm.estack.PushVal(keys[i])
+	}
+
+	vm.estack.PushVal(len(keys))
+	runVM(t, vm)
+
+	require.Equal(t, 1, vm.estack.Len())
+	packedItem := vm.estack.Pop().Item()
+	assert.IsType(t, &stackitem.Map{}, packedItem)
+
+	packedMap := packedItem.(*stackitem.Map)
+	require.Equal(t, 3, packedMap.Len())
+
+	checkKeyValue := func(key, expectedValue string) {
+		hasKey := packedMap.Has(stackitem.Make(key))
+		require.True(t, hasKey)
+		index := packedMap.Index(stackitem.Make(key))
+		res := packedMap.Value().([]stackitem.MapElement)
+		b, err := res[index].Value.TryBytes()
+		require.NoError(t, err)
+		assert.Equal(t, []byte(expectedValue), b)
+	}
+
+	checkKeyValue("duplicateKey", "value1")
+	checkKeyValue("uniqueKey", "value2")
+	checkKeyValue("anotherUniqueKey", "value4")
```

Comment:
Too complicated, and this check depends on `Index()` implementation. Replace it with `require.Equal(t, []stackitem.Item{stackitem.MapElement{...}, ...}, packedMap.Value())`.